docs/src/main/paradox/scala/http/common/caching.md

```diff
@@ -0,0 +1,88 @@
+# Caching
+
+Akka HTTP's caching support provides a lightweight and fast in-memory caching
+functionality based on futures. The primary use-case is the "wrapping" of an
+expensive operation with a caching layer that, based on a certain key of type
+`K`, runs the wrapped operation only once and returns the the cached value for
+all future accesses for the same key (as long as the respective entry has not
+expired).
+
+Akka HTTP comes with one implementations of the `Cache` API built on [Caffeine]
+featuring last-recently-used (LRU) cache eviction semantics with support for
+time-based entry expiration.
+
+ [Caffeine]: https://github.com/ben-manes/caffeine/
+
+To enable caching support add a library dependency onto:
+
+@@dependency [sbt,Gradle,Maven] {
+  group="com.typesafe.akka"
+  artifact="akka-http-caching_$scala.binary.version$"
+  version="$project.version$"
+}
+
+## Basic design
+
+The central idea of the cache API is to not store the actual values of type `T`
+themselves in the cache but rather the corresponding futures, i.e. instances of
+type @java[`CompletableFuture<T>`]@scala[`Future[T]`]. This approach has the
+advantage of nicely taking care of the thundering herds problem where many
+requests to a particular cache key (e.g. a resource URI) arrive before the first
+one could be completed. Normally (without special guarding techniques, like
+so-called "cowboy" entries) this can cause many requests to compete for system
+resources while trying to compute the same result thereby greatly reducing
+overall system performance. When you use an Akka HTTP cache the very first
+request that arrives for a certain cache key causes a future to be put into the
+cache which all later requests then "hook into". As soon as the first request
+completes all other ones complete as well. This minimizes processing time and
+server load for all requests.
+
+All Akka HTTP cache implementations implement the @java[@javadoc[`Cache`
+interface](akka.http.caching.javadsl.Cache)]@scala[@scaladoc[`Cache`
+class](akka.http.caching.scaladsl.Cache)], which allows you to interact with the
+cache.
+
+@@@ note
+
+Akka HTTP cache implementations must currently implement both the Java and Scala
+cache interface. As such, you are currently encouraged to implement new caches
+in Scala.
+
+@@@
+
+Along with the cache API, the routing DSL provides several @ref:[caching
+directives](../routing-dsl/directives/caching-directives/index.md) to use
+caching in your routes.
+
+## Last-frequently-used cache implementation
```

Comment:
This probably needs to clarify that it is a frequency-biased  implementation rather than recency-based as pointed out in https://github.com/akka/akka-http/pull/1503#discussion_r148699709